java/google/registry/flows/FlowMetadata.java

```diff
@@ -0,0 +1,24 @@
+package google.registry.flows;
+
+import com.google.auto.value.AutoValue;
+import google.registry.model.ImmutableObject;
+
+/** Object to allow setting and retrieving metadata information in flows. */
+@AutoValue
+public abstract class FlowMetadata extends ImmutableObject {
+
+  public abstract boolean superuser();
```

Comment:
I wasn't suggesting getHistoryEntry, getEntityChanges, etc (i.e. method names).  I'm suggesting a field name, which is, admittedly, a little bit confusing in the context of @AutoValue, in which you are creating fields, but they're defined using method syntax because that's how AutoValue works.  The field historyEntry is a history entry, the field entityChanges is the list of entity changes, and the field superuser would be a superuser (whatever object that would be).  isSuperuser makes it clear that the field is a boolean, not a superuser object itself.  "is" is a common prefix in our codebase on boolean fields that makes it clear without having to inspect the type that what's being referred to is a boolean, not a full object.  So "isSuperuser" is good, and that's what it already is in the flow.